extractor/src/main/java/com/example/extractor/Main.java

```diff
@@ -15,13 +15,20 @@ public class Main {
     private static final String PROJECT_PROPERTIES_FILE_PATH = "src/main/resources/" + PROJECT_PROPERTIES_FILENAME;
 
     public static void main(String[] args) throws Exception {
-        if (args.length < 1) {
-            throw new IllegalArgumentException("Path in repo must be provided as argument.");
+        if (args.length < 2) {
+            throw new IllegalArgumentException("Path in repo and configuration type must be provided as arguments.");
         }
 
-        // Get the path in repo from the command-line arguments
+        // Get the path in repo and configuration type from the command-line arguments
         String pathInRepo = args[0];
+        String configType = args[1].toLowerCase();
+
+        if (!configType.equals("treewalker") && !configType.equals("non-treewalker")) {
+            throw new IllegalArgumentException("Configuration type must be either 'treewalker' or 'non-treewalker'.");
```

Comment:
This is not ok.
You just need to load config and find treewalker in it in this case it is treewalker module. If no module with name Treewalker, it is not treewalker